pkgs/games/purpur/default.nix

```
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchurl, nixosTests, jre_headless }:
+let
+  version = "1.17.1";
+  build = "1428";
+in
+stdenv.mkDerivation {
+  pname = "purpur";
+  version = "${version}r${build}";
+
+  src = fetchurl {
+    url = "https://api.purpurmc.org/v2/purpur/${version}/${build}/download";
+    sha256 = "0sjrj67bgkn7ivab36aifds337ps2jajwk9628ilyhlwm1i4l089";
+  };
+
+  preferLocalBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin $out/lib/minecraft
+    cp -v $src $out/lib/minecraft/server.jar
+
+    cat > $out/bin/minecraft-server << EOF
+    #!/bin/sh
+    exec ${jre_headless}/bin/java \$@ -jar $out/lib/minecraft/server.jar nogui
+    EOF
+
+    chmod +x $out/bin/minecraft-server
```

This requires `makeWrapper` in `nativeBuildInputs`, but it's a bit cleaner:
```suggestion
    makeWrapper ${jre_headless}/bin/java $out/bin/minecraft-server \
      --add-flags "-jar $out/lib/minecraft/server.jar nogui"
```